flink-runtime/src/test/java/org/apache/flink/runtime/codegeneration/SortingTest.java

```diff
@@ -0,0 +1,617 @@
+		if (!this.memoryManager.verifyEmpty()) {
+			Assert.fail("Memory Leak: Some memory has not been returned to the memory manager.");
+		}
+		
+		if (this.memoryManager != null) {
+			this.memoryManager.shutdown();
+			this.memoryManager = null;
+		}
+	}
+
+	private InMemorySorter<Tuple2<Integer, String>> newSortBuffer(List<MemorySegment> memory) throws Exception {
+		return this.sorterFactory.createSorter(TestData.getIntStringTupleSerializer(), TestData.getIntStringTupleComparator(), memory);
+	}
+
+	@Test
+	public void testSortIntKeys() throws Exception {
+		final int NUM_RECORDS = 559273;
+
+		final int numSegments = MEMORY_SIZE / MEMORY_PAGE_SIZE;
+		final List<MemorySegment> memory = this.memoryManager.allocatePages(new DummyInvokable(), numSegments);
+
+		InMemorySorter<Tuple2<Integer, String>> sorter = newSortBuffer(memory);
+		TestData.TupleGenerator generator = new TestData.TupleGenerator(SEED, KEY_MAX, VALUE_LENGTH, KeyMode.RANDOM,
+			ValueMode.RANDOM_LENGTH);
+
+		// write the records
+		Tuple2<Integer, String> record = new Tuple2<>();
+		int num = 0;
+		do {
+			generator.next(record);
+			num++;
+		}
+		while (sorter.write(record) && num < NUM_RECORDS);
+
+		QuickSort qs = new QuickSort();
+		qs.sort(sorter);
+
+		MutableObjectIterator<Tuple2<Integer, String>> iter = sorter.getIterator();
+		Tuple2<Integer, String> readTarget = new Tuple2<>();
+
+		iter.next(readTarget);
+		int last = readTarget.f0;
+
+		while ((readTarget = iter.next(readTarget)) != null) {
+			int current = readTarget.f0;
+
+			final int cmp = last - current;
+			if (cmp > 0) {
+				Assert.fail("Next key is not larger or equal to previous key.");
+			}
+
+			last = current;
+		}
+
+		// release the memory occupied by the buffers
+		sorter.dispose();
+		this.memoryManager.release(memory);
+	}
+
+	@Test
+	public void testSortStringKeys() throws Exception {
+
+		@SuppressWarnings("unchecked")
+		List<MemorySegment> memory = createMemory();
+		TypeComparator<Tuple2<Integer, String>> accessors = TestData.getIntStringTupleTypeInfo().createComparator(new int[]{1}, new boolean[]{true}, 0, null);
+		InMemorySorter<Tuple2<Integer, String>> sorter = createSorter(TestData.getIntStringTupleSerializer(), accessors, memory);
+
+		TestData.TupleGenerator generator = new TestData.TupleGenerator(SEED, KEY_MAX, 5, KeyMode.RANDOM,
+			ValueMode.FIX_LENGTH);
+
+		Tuple2<Integer, String> record = new Tuple2<>();
+		do {
+			generator.next(record);
+		}
+		while (sorter.write(record));
+
+		QuickSort qs = new QuickSort();
+		qs.sort(sorter);
+
+		MutableObjectIterator<Tuple2<Integer, String>> iter = sorter.getIterator();
+		Tuple2<Integer, String> readTarget = new Tuple2<>();
+
+		iter.next(readTarget);
+		String last = readTarget.f1;
+
+		while ((readTarget = iter.next(readTarget)) != null) {
+			String current = readTarget.f1;
+
+			final int cmp = last.compareTo(current);
+			if (cmp > 0) {
+				Assert.fail("Next value is not larger or equal to previous value.");
```

Comment:
I can't think of any simple way to do this. `Assert.assertArrayEqual` doesn't work because of the reason that you mentioned.

I think you should just check that the final order is sorted, and that the same elements appear the same number of times.